### ISSUE TITLE
`watchTriggers` doesn't work with `..` in a glob path

### ISSUE THREAD

**Issue by japgolly** [sbt/sbt#6387]
`watchTriggers` doesn't work with `..` in a glob path.

When I add below to my build and run `~compile` it never detects my changes in my `.../scala-3` directory.

```scala
sbt.nio.Keys.watchTriggers +=
  file("/home/golly/projects/public/univeq/univeq/jvm/../shared/src/main/scala-3/").toGlob / ** / "*.scala"
```

Running with `sbt --debug` I can see that when a file changes, the watch machinery isn't triggered. No debug logging is printed at all.

However the `watchTriggers` setting to remove the `..` like below, makes it all work

```scala
sbt.nio.Keys.watchTriggers +=
  file("/home/golly/projects/public/univeq/univeq/shared/src/main/scala-3/").toGlob / ** / "*.scala"
```

So it seems that the watch machinery isn't handling the `..` for some reason.

## Side note

If you're wondering why this is an issue, I'm not manually writing out that full path; that was just the end of my debugging. As part of my work to upgrade libraries cross-built with Scala.JS to Scala 3, I added a snippet like this:

```scala
      unmanagedSourceDirectories in Compile ++= {
        val root   = (baseDirectory in Compile).value / ".."
        val shared = root / "shared" / "src" / "main"
        CrossVersion.partialVersion(scalaVersion.value) match {
          case Some((2, _))  => Seq(shared / "scala-2")
          case Some((3, _))  => Seq(shared / "scala-3")
          case _             => Nil
        }
      },
```

but watching wasn't working. I tried basically copy-pasting to provide the same setting to `watchTriggers` but as you can see there's a programmatic `..` required to navigate back to the `shared` directory.

**Comment by eatkins** [comment]
Yeah, I'm not shocked this didn't work. The relevant code for the conversion of a file or path to a glob is here: https://github.com/sbt/io/blob/707ea963a6f98ed1f9275b78e6a94b8dbb3571ef/io/src/main/scala/sbt/nio/file/Glob.scala#L500. I think the `..` would work if the path was normalized first.

I was going to suggest `getParentFile` as a workaround, but I see that you figured that out yourself with the linked PR.
